staking/cli/src/instructions.rs

```diff
@@ -870,6 +878,191 @@ pub fn update_y(rpc_client: &RpcClient, signer: &dyn Signer, y: u64) {
+            .as_ref(),
+    )
+    .unwrap();
+
+    pool_data
+        .publishers
+        .iter()
+        .enumerate()
+        .for_each(|(publisher_index, publisher)| {
+            if *publisher == Pubkey::default() {
+                return;
+            }
+            let publisher_stake_account_positions =
+                if pool_data.publisher_stake_accounts[publisher_index] == Pubkey::default() {
+                    None
+                } else {
+                    Some(pool_data.publisher_stake_accounts[publisher_index])
+                };
+
+            let publisher_stake_account_custody =
+                publisher_stake_account_positions.map(get_stake_account_custody_address);
+
+            let accounts = integrity_pool::accounts::AdvanceDelegationRecord {
+                delegation_record: get_delegation_record_address(*publisher, *positions),
+                payer: signer.pubkey(),
+                pool_config,
+                pool_data: pool_data_address,
+                pool_reward_custody: get_pool_reward_custody_address(pyth_token_mint),
+                publisher: *publisher,
+                publisher_stake_account_positions,
+                publisher_stake_account_custody,
+                stake_account_positions: *positions,
+                stake_account_custody: get_stake_account_custody_address(*positions),
+                system_program: system_program::ID,
+                token_program: spl_token::ID,
+            };
+
+            let data = integrity_pool::instruction::AdvanceDelegationRecord {};
+
+            let ix = Instruction {
+                program_id: integrity_pool::ID,
+                accounts:   accounts.to_account_metas(None),
+                data:       data.data(),
+            };
+
+
+            let mut transaction =
+                Transaction::new_with_payer(&[ix.clone()], Some(&signer.pubkey()));
+            transaction.sign(
+                &[signer],
+                rpc_client
+                    .get_latest_blockhash_with_commitment(CommitmentConfig::finalized())
+                    .unwrap()
+                    .0,
+            );
+            let res = rpc_client.simulate_transaction(&transaction).unwrap();
+            let reward_amount: u64 = u64::from_le_bytes(
+                BASE64_STANDARD
+                    .decode(res.value.return_data.unwrap().data.0)
+                    .unwrap()[..8]
+                    .try_into()
+                    .unwrap(),
+            );
+
+            if reward_amount < min_reward {
+                return;
+            }
+
+            println!(
+                "Advance delegation record for pubkey: {:?} publisher: {:?} with reward: {:?}",
+                positions.to_string(),
+                publisher,
+                reward_amount,
+            );
+
+            process_transaction(rpc_client, &[ix], &[signer]).unwrap();
```

Comment:
should we add `merge_delegation_positions` here

Reply:
N/A anymore